### PR TITLE
Update Framework Databinding How-To

### LINKED ISSUE #1860: WPF databinding .NET Framework how-to is REALLY bad - Rewrite it for .NET 8 and backport it
### Type of issue

Code doesn't work

### Description

This example is missing a lot of context and information. WPF for .NET 8 doesn't have this article but it should. Write a new How To for .NET 8  and then port it back replacing this one.

### Page URL

https://learn.microsoft.com/en-us/dotnet/desktop/wpf/data/how-to-create-a-simple-binding?view=netframeworkdesktop-4.8&viewFallbackFrom=netdesktop-8.0

### Content source URL

https://github.com/dotnet/docs-desktop/blob/main/dotnet-desktop-guide/framework/wpf/data/how-to-create-a-simple-binding.md

### Document Version Independent Id

a23a2ddb-9daf-0ecf-d830-963a4f4a5702

### Article author

@adegeo

---
[Associated WorkItem - 265721](https://dev.azure.com/msft-skilling/Content/_workitems/edit/265721)

### PR DESCRIPTION
**Pull request by adegeo**
## Summary

- Rewrite .NET Framework WPF databinding article.
- Recode C# and XAML to demonstrate an Employee data object and view.
- Updated .NET's WPF databinding overview to show its data object.

Fixes #1860

@IEvangelist Just checkout the wording on the non-Framework file as the content is 99% the same between both versions of the file. They're they same but slightly tuned for Framework and non-Framework for links.

**File in GitHub**: https://github.com/dotnet/docs-desktop/pull/1868/files#diff-6c1179ceeab49612218bf13bd09449ef751ea565ca28cd51c2754bf85187fd2c

**Preview**: https://review.learn.microsoft.com/en-us/dotnet/desktop/wpf/data/how-to-create-a-simple-binding?branch=pr-en-us-1868&view=netdesktop-7.0

---

#### Internal previews

| 📄 File | 🔗 Preview link |
|:--|:--|
| [dotnet-desktop-guide/framework/wpf/data/how-to-create-a-simple-binding.md](https://github.com/dotnet/docs-desktop/blob/1ec0ca3bf698bc612431911867abe1599420a18b/dotnet-desktop-guide/framework/wpf/data/how-to-create-a-simple-binding.md) | [How to: Create a Simple Binding](https://review.learn.microsoft.com/en-us/dotnet/desktop/wpf/data/how-to-create-a-simple-binding?branch=pr-en-us-1868&view=netframeworkdesktop-4.8) |
| [dotnet-desktop-guide/net/wpf/data/binding-declarations-overview.md](https://github.com/dotnet/docs-desktop/blob/1ec0ca3bf698bc612431911867abe1599420a18b/dotnet-desktop-guide/net/wpf/data/binding-declarations-overview.md) | [Binding declarations overview (WPF .NET)](https://review.learn.microsoft.com/en-us/dotnet/desktop/wpf/data/binding-declarations-overview?branch=pr-en-us-1868&view=netdesktop-7.0) |
| [dotnet-desktop-guide/net/wpf/data/how-to-create-a-simple-binding.md](https://github.com/dotnet/docs-desktop/blob/1ec0ca3bf698bc612431911867abe1599420a18b/dotnet-desktop-guide/net/wpf/data/how-to-create-a-simple-binding.md) | [dotnet-desktop-guide/net/wpf/data/how-to-create-a-simple-binding](https://review.learn.microsoft.com/en-us/dotnet/desktop/wpf/data/how-to-create-a-simple-binding?branch=pr-en-us-1868&view=netdesktop-7.0) |

### CODE REVIEW

File: dotnet-desktop-guide/framework/wpf/data/snippets/how-to-create-a-simple-binding/csharp/App.xaml.cs [hunk 1]
```diff
@@ -0,0 +1,18 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+using System.Configuration;
+using System.Data;
+using System.Linq;
+using System.Threading.Tasks;
+using System.Windows;
```
Comment (IEvangelist) [inline dotnet-desktop-guide/framework/wpf/data/snippets/how-to-create-a-simple-binding/csharp/App.xaml.cs:8]: Are all of these `using` needed?

File: dotnet-desktop-guide/framework/wpf/data/snippets/how-to-create-a-simple-binding/csharp/EmployeeView.xaml.cs [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+using System.Text;
+using System.Threading.Tasks;
+using System.Windows;
+using System.Windows.Controls;
+using System.Windows.Data;
+using System.Windows.Documents;
+using System.Windows.Input;
+using System.Windows.Media;
+using System.Windows.Media.Imaging;
+using System.Windows.Shapes;
```
Comment (IEvangelist) [inline dotnet-desktop-guide/framework/wpf/data/snippets/how-to-create-a-simple-binding/csharp/EmployeeView.xaml.cs:13]: `using` overload...why, so many?! 🤣 Probably comes from the template.

File: dotnet-desktop-guide/net/wpf/data/how-to-create-a-simple-binding.md [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+#customer intent: As a devleoper, I want to create a binding so that I can present information in a UI
+---
+
+# How to create a data binding (WPF .NET)
+
+This article describes how to create a binding XAML. The example uses a data object that represents an employee at a company. This data object is bound to a XAML window that uses `TextBlock` controls to list the employee's details. You'll create a UI that looks like the following image:
+
+:::image type="content" source="media/how-to-create-a-simple-binding/preview.png" alt-text="A WPF window that shows details about an employee, such as their first name, last name, title, hire date, and salary.":::
+
+To learn more about data binding, see [Data binding overview in WPF](data-binding-overview.md). To learn more about data binding, see [Data binding overview in WPF](data-binding-overview.md).
```
Comment (IEvangelist) [inline dotnet-desktop-guide/net/wpf/data/how-to-create-a-simple-binding.md:21]: Remove repeated sentence.

```suggestion
To learn more about data binding, see [Data binding overview in WPF](data-binding-overview.md).
```

File: dotnet-desktop-guide/net/wpf/data/how-to-create-a-simple-binding.md [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+1.  Replace the code with the following snippet:
+
+    :::code language="csharp" source="./snippets/how-to-create-a-simple-binding/csharp/Employee.cs":::
+    :::code language="vb" source="./snippets/how-to-create-a-simple-binding/vb/Employee.vb":::
+
+The employee data object is a simple class that describes an employee:
+
+- The first and last name of the employee.
+- The date the employee was hired.
+- The title employee has at the company.
```
Comment (IEvangelist) [inline dotnet-desktop-guide/net/wpf/data/how-to-create-a-simple-binding.md:37]:
```suggestion
- The employee's company title.
```

**Review by IEvangelist — APPROVED** [review]
LGTM, had a few suggestions.

File: dotnet-desktop-guide/framework/wpf/data/snippets/how-to-create-a-simple-binding/csharp/EmployeeView.xaml.cs (hunk 2 above)
Comment (adegeo) [reply dotnet-desktop-guide/framework/wpf/data/snippets/how-to-create-a-simple-binding/csharp/EmployeeView.xaml.cs:13]: Yep!

**Pull request merged**